workflows/src/test/java/com/r3/corda/lib/tokens/workflows/TokenBuilderTests.java

```diff
@@ -0,0 +1,132 @@
+                .withAmountValue(1)
+                .of(DigitalCurrency.getInstance("BTC"))
+                .issuedBy(aliceParty)
+                .heldBy(aliceParty)
+                .buildFungibleToken();
+    }
+
+    @Test
+    public void VaryingInputAmountTypesAreEquivalent() {
+        Amount<TokenType> intAmountTokenType = new TokenBuilder()
+                .withAmountValue(1)
+                .of(FiatCurrency.getInstance("USD"))
+                .buildAmountTokenType();
+
+        Amount<TokenType> doubleAmountTokenType = new TokenBuilder()
+                .withAmountValue(1.0)
+                .of(FiatCurrency.getInstance("USD"))
+                .buildAmountTokenType();
+
+        Amount<TokenType> bigDecimalAmountTokenType = new TokenBuilder()
+                .withAmountValue(BigDecimal.ONE)
+                .of(FiatCurrency.getInstance("USD"))
+                .buildAmountTokenType();
+
+        Amount<TokenType> longAmountTokenType = new TokenBuilder()
+                .withAmountValue(new Long(1))
+                .of(FiatCurrency.getInstance("USD"))
+                .buildAmountTokenType();
+
+        Amount<TokenType> total = intAmountTokenType
+                .plus(doubleAmountTokenType)
+                .plus(bigDecimalAmountTokenType)
+                .plus(longAmountTokenType);
+
+        assert(total.getQuantity() == AmountUtilities.of(4, FiatCurrency.getInstance("USD")).getQuantity());
+    }
+
+    @Test
+    public void AmountMayBeSetMoreThanOnce() throws Exception {
+        Amount<TokenType> amount = new TokenBuilder()
+                .withAmountValue(new Long(1))
+                .withAmountValue(2)
+                .of(FiatCurrency.getInstance("USD"))
+                .buildAmountTokenType();
+        assert (amount.getQuantity() == 200L);
+    }
+
+    @Test
+    public void UnableToRetrieveAmountTokenTypeWithoutTokenType() throws Exception {
+        try {
+            new TokenBuilder()
+                    .withAmountValue(new Long(1))
+                    .buildAmountTokenType();
+            assert(false);
+        } catch(Exception ex) {
+            assert(ex instanceof TokenBuilderException);
+            assert(ex.getLocalizedMessage().equals("A Token Type has not been provided to the builder."));
+        }
+    }
+
+    @Test
+    public void UnableToRetrieveAmountIssuedTokenTypeWithoutIssuer() throws Exception {
+        try {
+            new TokenBuilder()
+                    .withAmountValue(new Long(1))
+                    .of(FiatCurrency.getInstance("USD"))
+                    .buildAmountIssuedTokenType();
+            assert(false);
+        } catch(Exception ex) {
+            assert(ex instanceof TokenBuilderException);
+            assert(ex.getLocalizedMessage().equals("An token issuer has not been provided to the builder."));
+        }
+    }
+
+    @Test
+    public void UnableToRetrieveFungibleTokenWithoutHolder() throws Exception {
```

Comment:
It's better to be more specific about types of exceptions the method throws. For example here would be better to use `throws TokenBuilderException`

Reply:
Fixed! I hadn't added the @Throws(TokenBuilderException::class) to the Kotlin function which was causing a compilation error.